### PR TITLE
Add HID article

### LINKED ISSUE #3571: content: connecting to uncommon devices with webhid [2020-08-24]
* [draft](https://docs.google.com/document/d/1hxnbSvMZUOiiz4XQEKSeht-wsUKgJdt2cZYk4azchso/edit#)
* [thread](https://groups.google.com/a/google.com/g/webdev-content/c/AbH9H6UuwOg)
* next step: @jpmedley to review

### PR DESCRIPTION
**Pull request by beaufortfrancois**
Fixes #3571

Changes proposed in this pull request:

- Add the article dedicated to the WebHID API.
- Add "WebHID" to the "New capabilities status" page.
- Add "hid" to the list of features allowed in glitch embeds.

Live preview: 
- https://deploy-preview-3860--web-dev-staging.netlify.app/hid/
- https://deploy-preview-3860--web-dev-staging.netlify.app/fugu-status/

### CODE REVIEW

File: src/site/content/en/blog/hid/index.md [hunk 1]
```diff
@@ -0,0 +1,478 @@
...
+});
+```
+
+## Dev Tips {: #dev-tips }
+
+Debugging HID in Chrome is easy with the internal page, `chrome://device-log`
+where you can see all HID and USB device related events in one single place.
+
+<figure class="w-figure">
+  <img src="./device-log-page-screenshot.jpg" class="w-screenshot" alt="Screenshot of the internal page to debug HID">
```
Comment (jpmedley) [inline src/site/content/en/blog/hid/index.md:374]:
```suggestion
  <img src="./device-log-page-screenshot.jpg" class="w-screenshot" alt="Screenshot of the internal page to debug HID.">
```

File: src/site/content/en/blog/hid/index.md [hunk 2]
```diff
@@ -0,0 +1,478 @@
...
+  <figcaption class="w-figcaption">Internal page in Chrome to debug HID</figcaption>
```
Comment (jpmedley) [inline src/site/content/en/blog/hid/index.md:375]:
```suggestion
  <figcaption class="w-figcaption">Internal page in Chrome to debug HID.</figcaption>
```

File: src/site/content/en/blog/hid/index.md [hunk 3]
```diff
@@ -0,0 +1,478 @@
...
+API is primarily gated by a permission model that grants access to only a single
+HID device at a time. In response to a user prompt, the user must take active
+steps to select a particular HID device.
+
+To understand the security tradeoffs, check out the [Security and Privacy
+Considerations] section of the WebHID spec.
+
+On top of this, Chrome inspects the usage of each top-level collection and if a
+top-level collection has a protected usage (.e.g. generic keyboard, mouse), then
+a website won't be able to send to and receive from any reports defined in that
```
Comment (jpmedley) [inline src/site/content/en/blog/hid/index.md:396]:
```suggestion
a website won't be able to send and receive any reports defined in that
```

Comment (jpmedley) [reply src/site/content/en/blog/hid/index.md:396]: This is not a suggestion to omit needless words. This is a grammatical problem I should have caught before.

**Review by jpmedley — CHANGES_REQUESTED** [review]
François,

This looks pretty good. The requested changes fall into two categories: text in figures, which I didn't have access to in the first review; a few thins I should have caught before. Sorry about the latter.

Joe

File: src/site/content/en/blog/hid/index.md [hunk 4]
```diff
@@ -0,0 +1,478 @@
+---
+title: Connecting to uncommon HID devices
+subhead: |
+  The WebHID API allows websites to access alternative auxiliary keyboards and exotic gamepads.
+authors:
+  - beaufortfrancois
+date: 2020-09-11
+updated: 2020-09-11
```
Comment (jpmedley) [inline src/site/content/en/blog/hid/index.md:8]:
```suggestion
date: 2020-09-14
updated: 2020-09-14
```

**Review by jpmedley — CHANGES_REQUESTED** [review]

File: src/site/content/en/blog/hid/index.md [hunk 5]
```diff
@@ -0,0 +1,478 @@
...
+
+A HID usage is a numeric value referring to a standardized input or output.
+Usage values allow a device to describe the intended use of the device and the
+purpose of each field in its reports. For example, one is defined for the left
+button of a mouse. Usages are also organized into usage pages, which provide an
+indication of the high-level category of the device or report.
+
+## Using the WebHID API {: #use }
+
+### Enabling support during the origin trial phase
```
Comment (scheib) [inline src/site/content/en/blog/hid/index.md:121]: I suggest the article instruct flag usage first (simpler) and then why and how to use an Origin Trial.

Comment (beaufortfrancois) [reply src/site/content/en/blog/hid/index.md:120]: @petele @tomayac What do you think of this suggestion? Shall we change our fugu doc template?

Comment (tomayac) [reply src/site/content/en/blog/hid/index.md:120]: 👍 I'm not opposed. Most developers, even the ones who end up going the OT route, will likely first test locally with a flag flipped.

Comment (beaufortfrancois) [reply src/site/content/en/blog/hid/index.md:120]: I've noticed afterwards you actually moved it in your recent articles. Let's update our template then.

Comment (beaufortfrancois) [reply src/site/content/en/blog/hid/index.md:126]: Fixed in https://github.com/GoogleChrome/web.dev/pull/3860/commits/61184ac69b90e61c13f54d06ac074010b077b1a5

**Comment by beaufortfrancois** [comment]
@jpmedley Thanks for the second review. Your feedback has been addressed in https://github.com/GoogleChrome/web.dev/pull/3860/commits/b598538e2fd2e3e63bad8901fd08731c80b05e0e

File: src/site/content/en/blog/hid/index.md [hunk 6]
```diff
@@ -0,0 +1,477 @@
...
+
+### Listen to connection and disconnection {: #connection-disconnection }
+
+When the website has been granted permission to access a HID device, it can
+actively receive connection and disconnection events by listening to `"connect"`
+and `"disconnect"` events.
+
+```js
+navigator.hid.addEventListener("connect", event => {
+  // TODO: Automatically open event.device or warn user a device is available.
```
Comment (jpmedley) [inline src/site/content/en/blog/hid/index.md:359]:
```suggestion
  // Automatically open event.device or warn user a device is available.
```

File: src/site/content/en/blog/hid/index.md [hunk 7]
```diff
@@ -0,0 +1,477 @@
...
+});
+
+navigator.hid.addEventListener("disconnect", event => {
+  // TODO: Remove |event.device| from the UI.
```
Comment (jpmedley) [inline src/site/content/en/blog/hid/index.md:363]:
```suggestion
  // Remove |event.device| from the UI.
```

**Review by jpmedley — CHANGES_REQUESTED** [review]
François,

There are still open issues from my previous review.

File: src/site/content/en/blog/hid/index.md [hunk 8]
```diff
@@ -0,0 +1,477 @@
...
+
+A HID usage is a numeric value referring to a standardized input or output.
+Usage values allow a device to describe the intended use of the device and the
+purpose of each field in its reports. For example, one is defined for the left
+button of a mouse. Usages are also organized into usage pages, which provide an
+indication of the high-level category of the device or report.
+
+## Using the WebHID API {: #use }
+
+### Enabling via chrome://flags
```
Comment (jpmedley) [inline src/site/content/en/blog/hid/index.md:120]: This probably does need to be moved, but it needs to consistent between articles. Please move this back.

Comment (beaufortfrancois) [reply src/site/content/en/blog/hid/index.md:119]: As communicated offline, I'll keep this way and will update other articles in a separate PR.

**Comment by beaufortfrancois** [comment]
> François,
> 
> There are still open issues from my previous review.

Sorry for that. I've addressed your previous feedback and new one at https://github.com/GoogleChrome/web.dev/pull/3860/commits/bc55dbad698b69a43971ec4b1d0c184e08db7ffc

File: src/site/content/en/blog/hid/index.md (hunk 8 above)
Comment (beaufortfrancois) [reply src/site/content/en/blog/hid/index.md:118]: https://github.com/GoogleChrome/web.dev/pull/3875

**Review by jpmedley — APPROVED** [review]

**Pull request merged**